### ISSUE TITLE
Internal public functions

### ISSUE THREAD

**Issue by mdrost** [clue/php-term-react#21]
In `ControlCodeParser` I see some internal public functions. They are IMHO public because `callable arrays` cannot contains private functions. However current required PHP version is >=5.3 so anonymous functions are available. I suggest to make that functions private and pass them to `EventEmmiter` as anonymous functions.

ps. In PHP 5.3 anonymous functions need `$that = $this` hack, so I suggest to bump version to 5.4.

**Comment by clue** [comment]
Thanks for filing this issue as I think you're raising a very valid point :+1: 

These methods are marked for internal use only, so they *must not* be used from outside of this class which implies this is not something consumers of this package would have to care about. This is in fact required because they are called from outside of this class via the `EventEmitter`. This could in fact be avoided by wrapping them in an anonymous function – except for PHP 5.3 which can not call any private methods from within a closure. We *may* do all this and would then have to ditch PHP 5.3 support.

Long story short, how much would be gained by this?

**Comment by mdrost** [comment]
To call private function inside anonymous function in PHP 5.3 you need little hack:

```
$that = $this;
function () use ($that) {
    $that->call();
}
```

**Comment by clue** [comment]
Unfortunately, this doesn't work in legacy PHP < 5.4 because the closure is technically a `Closure` object and thus always outside of the calling scope and thus has no access to a private method from the calling scope.

**Comment by mdrost** [comment]
Oh, I made some tests, you are right. `$that = $this` only works with public functions.

**Comment by clue** [comment]
Do you still feel there's something that needs to be done here or is this resolved? :)

**Comment by mdrost** [comment]
I could only suggest bump version to [supported version 5.6](http://php.net/supported-versions.php) :P

**Comment by clue** [comment]
I agree that this is *reasonable*, but is much gained by doing this? :)

**Comment by clue** [comment]
This hasn't received any input in a while and I believe this has been answered, so I'm closing this for now. Please come back with more details if this problem persists and we can reopen this :+1:
